vendor/go.etcd.io/etcd/etcdserver/server.go

```diff
@@ -2670,6 +2675,15 @@ func (s *EtcdServer) Logger() *zap.Logger {
 
 // IsLearner returns if the local member is raft learner
 func (s *EtcdServer) IsLearner() bool {
+	tombstoneFile := filepath.Join(s.Cfg.DataDir, "tombstone")
```

Comment:
This seems pretty unlikely to be pulled upstream. I thought we were going to go with a unique exit code?

For RKE2, I think we should start a goroutine to watch or poll the etcd container status using the containerd client API, and create the file if etcd exits with the appropriate code.

Reply:
I dont think we need a unique exit code and yes that wont be pulled to upstream probably,we still need a file since in rke2 there is no way to check the pod exit status, because when the node is deleted the pod is automatically deleted, so it doesnt really matter

Reply:
You're correct that we can't check the static pod value since we can't guarantee that the static pod is currently sync'd to the cluster. We'd probably have to see if we can get it from the kubelet or containerd.

Reply:
I think containerd is the way to go, in any case  we need to settle on either a tombstone file or unique exit  code or both